python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -77,6 +84,13 @@ def __init__(
         self.client: tcpip.Client | None = None
         self.read_loop_task = utils.make_done_future()
 
+        self.mock_data_server: MockSiglentSSA3000xDataServer | None = None
+
+        self.number_of_truncated_data_read = 0
+
+        # Number of consecutive read timeouts encountered.
+        self.num_consecutive_read_timeouts = 0
```

Comment:
Please also reset this in the `connect` method (in both data clients). Thank you for resetting it after a successful read, but in addition it is important to clear it (and all other error accumulators) when you connect, so when we somebody re-enables a ESS, it starts with a clean slate.

A thought for a future ticket is to have a base class for "data clients with a read loop". That could centralize the error handling for read errors and timeouts.